app/services/distribution_itemized_breakdown_service.rb

```diff
@@ -0,0 +1,81 @@
+class DistributionItemizedBreakdownService
+  #
+  # Initializes the DistributionItemizedBreakdownService whoms
+  # purpose to construct a itemized breakdown of items distributed
+  # and what is left on-hand currently (at the time of running)
+  #
+  # @param organization [Organization]
+  # @param distribution_ids [Array<Integer>]
+  # @return [DistributionItemizedBreakdownService]
+  def initialize(organization:, distribution_ids:)
+    @organization = organization
+    @distributions = organization.distributions.where(id: distribution_ids).includes(line_items: :item)
+  end
+
+  #
+  # Returns a hash containing the itemized breakdown of
+  # what was distributed.
+  #
+  # @return [Hash]
+  def fetch
+    items_distributed = fetch_items_distributed
+
+    # Inject the "onhand" data
+    items_distributed.map! do |item|
+      item_name = item[:name]
+
+      item.merge({
+        current_onhand: current_onhand_quantities[item_name],
+        onhand_minimum: current_onhand_minimums[item_name],
+        below_onhand_minimum: current_onhand_quantities[item_name] < current_onhand_minimums[item_name]
+      })
+    end
+
+    items_distributed.sort_by { |item| -item[:distributed] }
+  end
+
+  #
+  # Returns a CSV string representation of the itemized breakdown of
+  # what was distributed
+  #
+  # @return [String]
+  def fetch_csv
+    convert_to_csv(fetch)
+  end
+
+  private
+
+  attr_reader :organization, :distributions
+
+  def current_onhand_quantities
+    @current_onhand_quantities ||= organization.inventory_items.group("items.name").sum(:quantity)
+  end
+
+  def current_onhand_minimums
+    @current_onhand_minimums ||= organization.inventory_items.group("items.name").maximum("items.on_hand_minimum_quantity")
+  end
+
+  def fetch_items_distributed
+    item_distribution_hash = distributions.each_with_object({}) do |d, acc|
+      d.line_items.each do |i|
+        key = i.item.name
+        acc[key] ||= {distributed: 0}
+        acc[key][:distributed] += i.quantity
+      end
+    end
+
+    item_distribution_hash.inject([]) do |acc, key_value|
+      acc << {name: key_value[0]}.merge(key_value[1])
+    end
```

Comment:
I'm having a hard time following this. By the end of the previous block it looks like we should have an object of the format
```ruby
{
  "item name" => { distributed: 999 }
}
```

But then this block seems to turn it into
```ruby
[{name: "item name", distributed: 999}]
```

Is that correct? If so, I'd find it a bit easier to parse if this used `map`:
```ruby
item_distribution_hash.map { |k, v| v.merge(name: k) }.values
```

Reply:
Yeh I figured it was a bit complicated when i wrote it out. I think your suggestion is way cleaner!

Reply:
Actually, this was sufficient :)
```
item_distribution_hash.map { |k, v| v.merge(name: k) }
```